src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -758,6 +762,19 @@ class PostgresEntityDataQueryService(
         return WriteEvent(System.currentTimeMillis(), numUpdates)
     }
 
+    private fun lockEntitiesForUpdate(
+            connection: Connection,
+            idsArr: java.sql.Array,
+            partition: Int,
+            partitionVersion: Int) {
+        // Acquire entity key id locks
+        val rowLocks = connection.prepareStatement(lockEntitiesSql)
```

Comment:
As a safety check it might be a good idea to ensure that connection.autoCommit = false here, because locks are released at end of transaction.

Reply:
This way we can catch unexpected invocations.